lib/resultuploaders/dropbox/dropbox.rb

```diff
@@ -20,20 +20,25 @@ def initialize(project)
       @logger = project.logger
       @repo = project.config['repository']
 
-      @config = Json.read_json(CONFIG_JSON, @logger)
-
-      @client_id = ENV.fetch('AUTOHCK_DROPBOX_CLIENT_ID')
-      @client_secret = ENV.fetch('AUTOHCK_DROPBOX_CLIENT_SECRET')
-
-      @action_retries = @config['action_retries']
-      @action_retry_sleep = @config['action_retry_sleep']
+      load_config
 
       @authenticator = DropboxApi::Authenticator.new(@client_id, @client_secret)
 
       @dropbox = nil
       @url = nil
     end
 
+    def html_url; end
+
+    def load_config
+      @config = Json.read_json(CONFIG_JSON, @logger)
+      @action_retries = @config['action_retries']
+      @action_retry_sleep = @config['action_retry_sleep']
+
+      @client_id = ENV.fetch('AUTOHCK_DROPBOX_CLIENT_ID')
+      @client_secret = ENV.fetch('AUTOHCK_DROPBOX_CLIENT_SECRET')
```

Comment:
`@client_id` and `@client_secret` are somewhat out of place; they are not from `CONFIG_JSON`. You may put them after `load_config`, or just abandon this change to introduce `load_config` method; I suspect this method is introduced to suppress the complexity warning caused by `@html_url` assignment, but `html_url` implementation is now stateless so it's no longer necessary.

Reply:
1. You are right `load_config` is not required for now.
2. `@client_id` and `@client_secret` are also configurations of the DropBox client, but these are secret values, so they pleased into ENV instead of `config.json`

Reply:
> 1. You are right `load_config` is not required for now.
> 
>     2. `@client_id` and `@client_secret` are also configurations of the DropBox client, but these are secret values, so they pleased into ENV instead of `config.json`

But the data loaded from a file is what is named `@config`. Perhaps `@config` is what should be renamed.

Reply:
Fixed